core/state_processor_rip7560.go

```diff
@@ -142,11 +145,27 @@ func BuyGasRip7560Transaction(st *types.Rip7560AccountAbstractionTx, state vm.St
 	}
 
 	if have, want := state.GetBalance(*chargeFrom), balanceCheck; have.Cmp(want) < 0 {
-		return fmt.Errorf("%w: address %v have %v want %v", ErrInsufficientFunds, chargeFrom.Hex(), have, want)
+		return nil, fmt.Errorf("%w: address %v have %v want %v", ErrInsufficientFunds, chargeFrom.Hex(), have, want)
 	}
 
-	state.SubBalance(*chargeFrom, mgval, 0)
-	return nil
+	state.SubBalance(*chargeFrom, preCharge, 0)
+	return preCharge, nil
+}
+
+// refund the transaction payer (either account or paymaster) with the excess gas cost
```

Comment:
I now notice that `refundGas` in `state_transition.go` is not modified for RIP-7560, meaning we probably currently "refund" somewhere (entrypoint address?) ether for unspent gas on every frame.

Reply:
right. created a separate issue.